halo2_frontend/src/plonk/circuit/constraint_system.rs

```diff
@@ -404,6 +410,14 @@ impl<F: Field> ConstraintSystem<F> {
                 if table.contains_simple_selector() {
                     panic!("expression containing simple selector supplied to lookup argument");
                 }
+                // this check ensures that one dedicated `Fixed` column or `Selector` is used
+                // for enabling the real table rows of the column, which is used as `TableColumn` role.
+                // otherwise, we get the soundness error, like https://github.com/privacy-scaling-explorations/halo2/issues/335
+                if table.degree() < 2 {
```

Comment:
sorry if im wrong, but i don't understand how the degree should be more than 2?
shouldn't it be the opposite? we want it to be <2
https://github.com/privacy-scaling-explorations/halo2/blob/b4d1c4c9c2cd4fdeab2b66b6872cdabbf700fcc9/halo2_frontend/src/plonk/circuit/expression.rs#L869

you say in the comment that you expect the column to be "Fixed column or Selector" but I thought the purpose of `lookup_any` was to allow a lookup into Advice column. If we're looking up into a Fixed column, we might as well use regular `lookup()`

Reply:
> shouldn't it be the opposite?  

No, the degree of `table expression` should be 2.
In other words, it should be like following:
```
meta.lookup_any(|cells| {
  
  let default = Expression::constant(F::ONE);
  let s_lookup = meta.query_selector(...);
  let to_lookup = meta.query_advice(...);
  let s_ltable = meta.query_selector(...);
  let ltable = meta.query_advice(...);
  vec![
    (
      s_lookup * to_lookup + (Expression::constant(F::ONE) - s_lookup) * default, 
      s_ltable * ltable
    )
  ]
}
```
This means, when using `lookup_any`, the advice/fixed column used for table role should be accompanied with another fixed column or selector to indicate which rows are used as table rows.

In the above example, the degree of `s_ltable * ltable` is 2. Since the degree of `selector` & `advice` column is 1, and the degree of its product is sum of degrees.

> you say in the comment that you expect the column to be "Fixed column or Selector" but I thought the purpose of lookup_any was to allow a lookup into Advice column.
I meant that extra fixed column or selector SHOULD be accompanied with table column(in `lookup_any`, the advice column).